src/Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs

```diff
@@ -159,15 +140,16 @@ type FSharpKernel() as this =
         handleRequestCompletion command context |> Async.StartAsTask :> Task
 
     override _.TryGetVariable<'a>(name: string, [<Out>] value: 'a byref) =
-        match this.GetCurrentVariable(name) with
-        | Some(cv) ->
-            value <- cv.Value :?> 'a
+        match script.Value.Fsi.TryFindBoundValue(name) with
+        | Some cv ->
+            value <- cv.Value.ReflectionValue :?> 'a
             true
-        | None ->
+        | _ ->
             false
 
     override _.SetVariableAsync(name: string, value: Object) : Task = 
-        raise (NotImplementedException())
+        script.Value.Fsi.AddBoundValue(name, value) |> ignore
```

Comment:
This will return a `FSharpErrorInfo` array. Should you report potential errors that occur?